### ISSUE TITLE
Append new headers to the client when reconnecting

### ISSUE THREAD

**Issue by Bowser65** [TooTallNate/Java-WebSocket#865]
**Is your feature request related to a problem? Please describe.**
I'm developing a client for an app that accepts a Resume-Id header when reconnecting for resuming a session. For now it looks like it's impossible to alter the list of headers which is a requirement for that kind of stuff

**Describe the solution you'd like**
Allow headers to be added, maybe only when socket is not connected.

**Describe alternatives you've considered**
It can be a .addHeader function, or a hook "onBeforeConnect" to add headers there

**Comment by haruntuncay** [comment]
If this is planned, I would like to help.

**Comment by marci4** [comment]
Maybe simple adding a setter would be enough?

**Comment by haruntuncay** [comment]
We could add a method to put additional headers and it could initialize `private Map<String,String> headers;` variable, if not initialized yet, and then just add the value. The values in the headers map would automatically be used with each call to `connect()` and `reconnect()`. 

I also added a method to remove a header since a user may not want to send every header on every call to `reconnect()`.
Btw, I actually have a commit ready for this, but I don't want that commit to go in to the pending PR, since that PR covers something entirely unrelated. I will send a new PR for this issue once the pending one is merged/closed.

**Comment by NoahAndrews** [comment]
@haruntuncay You can submit multiple PRs at once, you just need to submit them from different branches. In general it's a good idea to work on code in feature branches, rather than your master branch like you did with the currently open PR.

**Comment by haruntuncay** [comment]
> 
> 
> @haruntuncay You can submit multiple PRs at once, you just need to submit them from different branches. In general it's a good idea to work on code in feature branches, rather than your master branch like you did with the currently open PR.

Thank you for the notice. I wasn't aware that we could submit multiple PRs simultaneously.

**Comment by marci4** [comment]
Resolved by #868
